Modern theme update

babaissarkar opened the pull request:
* Updates various aspects based on feedback received in #9104
* Based on @Dalas121's design https://github.com/wesnoth/wesnoth/pull/9104#issuecomment-2294991431 but not an exact copy.
* Adds definitions for various highly used controls (but not everything)
* UI visible name for the themes have been changed: "Classic" -> "Parchment" and "Modern" -> "Celes"
* The Classic/Parchment theme remains the same except for the name change and is still default.
* A preference key + checkbox and a close button in tips panel has been added to show/hide it.
(Note: this theme does not provide equivalents for all existing widget definitions, just a minimum baseline.)

![Screenshot from 2024-12-29 19-10-01](https://github.com/user-attachments/assets/601d1e5f-4a0f-4dd6-b9e2-34bd9dc29e3c)
![Screenshot from 2025-01-08 22-55-44](https://github.com/user-attachments/assets/f8c6aef4-92d3-458b-b578-6af72bc69b2d)
![Screenshot from 2025-01-08 22-55-37](https://github.com/user-attachments/assets/f758bb98-c23d-41cd-a4a3-a65339077e7a)

Pentarctagon commented:
Is there a reason for the name change?

Wedge009 commented:
I think I requested that it not merely be called 'classic' vs 'modern', because what is 'modern' will eventually itself become classic by definition if not in name, which can cause confusion.

Pentarctagon commented:
"Parchment (Classic)" still says it's classic though. Blue Sky is also not really descriptive since, well, the classic UI is also really blue.

babaissarkar commented:
> "Parchment (Classic)" still says it's classic though. Blue Sky is also not really descriptive since, well, the classic UI is also really blue.

I can drop "Classic" from Parchment UI name, and change Blue Sky to "Celes", will that do? (Other possible names are fine with me)

Wedge009 commented:
> "Parchment (Classic)" still says it's classic though.

That concerns me less than having 'modern' in a name. I suppose I don't really mind what it's called as long as we settle on something that's beyond these generic terms.

Pentarctagon commented:
> I can drop "Classic" from Parchment UI name, and change Blue Sky to "Celes", will that do? (Other possible names are fine with me)

I'm fine with those, at least.

Vultraz commented:
Nice progress! I do think the sidebar looks rather bare without the logo to head it, and having the three big elements separate (the sidebar, the logo, and the help area) draws the eye in three opposite directions at first glance. I'd recommend keeping the logo (at the same size) at the head of the column, and having it break out of the edges (probably need a `stacked_widget`).

I'd also drop the outlines around the menu button text. It's just a bit much. There's also something subtly off about the spacing, which I've been trying to pin down... For one, the text isn't vertically centered in its little rectangle, and sits much closer to the top than the bottom. 🤔

babaissarkar commented:
> > "Parchment (Classic)" still says it's classic though. Blue Sky is also not really descriptive since, well, the classic UI is also really blue.
> 
> I can drop "Classic" from Parchment UI name, and change Blue Sky to "Celes", will that do? (Other possible names are fine with me)

Renamed to "Celes" and dropped "Classic" from Parchment. Screenshot updated.

knyghtmare commented:
The updated Main Menu UI looks amazing!

Vultraz commented:
Why the change from window/ to dialog/?

babaissarkar commented:
> Why the change from window/ to dialog/?

Consistency with source code folders.

Pentarctagon commented:
What's going on with
![image](https://github.com/user-attachments/assets/24f41a4c-3957-4b5d-88c5-255a31013b00)

babaissarkar commented:
> What's going on with ![image](https://private-user-images.githubusercontent.com/3030250/398269822-24f41a4c-3957-4b5d-88c5-255a31013b00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUwMDkxNTYsIm5iZiI6MTczNTAwODg1NiwicGF0aCI6Ii8zMDMwMjUwLzM5ODI2OTgyMi0yNGY0MWE0Yy0zOTU3LTRiNWQtODhjNS0yNTVhMzEwMTNiMDAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMjRUMDI1NDE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjRiYzBiM2UwZDZjMDg2N2U0ZDQyMDJlNzA4NmE0YjE1MjU0YzRjYjhlN2E0Yzk1MTFlOWRmMzZkZDNkZDBlMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.qwb8ZxjxKT73TcsBYILFZaSKIAo6Pt3EDo5YGU4XZcA)

Details? What is the issue?

Pentarctagon commented:
Yeah, I meant the border.

Vultraz commented:
I'm sorry for taking so long getting to this PR. Looking over the screenshots (assuming they represent the changes as they currently stand), there are unfortunately several big issues that prevent this from being merged as-is.

While the layout changes to the main menu are generally fine (save for the small logo, which absolutely must be adjusted), the broader UI changes as showcased in Prefs badly clash with the existing UI design and introduce several elements that make the game feel dated. (I've also done some tweaks to your existing main menu design locally, and - as discussed on Discord - I honestly think the readability and visibility complaints can be addressed with a much less drastic change. Especially with time to get used to what you did so far, it's a much more solid redesign than I initially thought.)

Onto these changes.

First, buttons: I really like the minimalist blue buttons you currently have on master. They're brighter, cleaner, and more modern than our old design, making the new menu feel new while not demanding too much of the player's attention. These buttons, on the other hand, feel far more like they're looking to emulate early 2000's, XP era skeuomorphic design. They look out of place against the flat, semitransparent tips box, and the ornate decoration makes them feel too "busy", for lack of a better word.

There are also issues with the dialog example. First and foremost, both the default button and the dropdown menu (especially the former) have visible vertical gradient banding which, unless it's just a screenshot artifact, is unacceptable for any UI in 2024.

Second, lack of visual cohesion. Checkboxes are surrounded by a bright yellow border, dropdown menus are shorter than buttons and feature non-recessed borders on the corners, and regular buttons feature protruding corner borders. Regular buttons are also narrower than menu buttons. The differing border sizes also means the visual alignment of text and elements is way off.

Third, there's also a major clash between the old and new visual styles. I know you mentioned this is an incomplete redesign and presumably more PRs will follow, but I would be very hesitant about committing to this new, more ornate design language without getting it right from the start. Going full-on ornamentation is probably not the way to go either, so resolving these inconsistencies (such as the minimal window borders vs the new controls) cannot just me a matter of adding more visual complexity.

I realize you've put a lot of work into implementing Dalas' proposals (and my own, for that matter), but I don't think the current product is ready. Some of these changes can definitely be added to master right now, though: The ability to hide the tips panel, for one, as well as the improved alignment of the language button with the tips panel controls and the new positioning of the version label.

Vultraz reviewed (changes requested):
See above comment.

babaissarkar commented:
@Vultraz

Thanks for the detailed review.

1. Can the current titlescreen buttons in master be used as the default button definition? I'm fine with that. (see the About/Language buttons)
2. Note that I actually deleted a lot of ornamentation from @Dalas121's original design. Regardless, I'm willing cleanup a bit more. (I don't get what you mean by "XP era, skeuomorphic" design.)
3. What's the vertical banding? You mean the pseudo 3d look of the buttons? Are you suggesting a more "material" style look?

babaissarkar commented:
@Dalas121 you have anything to say about Vultraz's comment, given that it is still your design with slight changes? (I just coded it)

Pentarctagon commented:
If you look at the Apply button in the first screenshot for example, there's (horizontal) banding.

babaissarkar commented:
> If you look at the Apply button in the first screenshot for example, there's (horizontal) banding.

![Screenshot from 2024-12-24 21-44-57](https://github.com/user-attachments/assets/9d17d30a-2620-4dec-9b2e-7dab3f99498f)
![Screenshot from 2024-12-24 21-44-45](https://github.com/user-attachments/assets/ed3c4656-ede3-4041-a9e1-42167fc7a4fb)

Is it these two parts?

Here is the actual background graphics I used:
![background](https://github.com/user-attachments/assets/2a68b233-39c3-4f7e-bd8d-c96574b0a1c8)
![background-active](https://github.com/user-attachments/assets/2f2d8195-5d6c-4276-a784-2168dd4bc686)
![background-pressed](https://github.com/user-attachments/assets/c8b08f30-f206-42b3-9acd-acc0abbe2362)

Is it visible here?

Dalas121 commented:
> @Dalas121 you have anything to say about Vultraz's comment, given that it is still your design with slight changes? (I just coded it)

As far as my personal preference, I generally prefer minimalist web-style UI elements - I have some background in that area and like the aesthetic. I'm not a UI expert though, so I'll defer to the two of you on specifics of the style. I'm available to create mockups for any style we decide on.

However, for future UI projects _(such as the in-scenario UI, which IMO is the most in need of improvment)_, it's probably best if we vet mockups with all involved developers before working on the implementation.

babaissarkar commented:
> However, for future UI projects _(such as the in-scenario UI, which IMO is the most in need of improvment)_, it's probably best if we vet mockups with all involved developers before working on the implementation.

Definitely, I thought Vultraz and other agreed with that design I linked. Wouldn't have tried otherwise.

Pentarctagon commented:
I mean the whole button:
![Screenshot from 2024-12-24 10-38-21](https://github.com/user-attachments/assets/4f008768-b269-44a6-82a8-84fb80cae756)

babaissarkar commented:
> I mean the whole button: ![Screenshot from 2024-12-24 10-38-21](https://private-user-images.githubusercontent.com/3030250/398447655-4f008768-b269-44a6-82a8-84fb80cae756.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUwNjA4OTgsIm5iZiI6MTczNTA2MDU5OCwicGF0aCI6Ii8zMDMwMjUwLzM5ODQ0NzY1NS00ZjAwODc2OC1iMjY5LTQ0YTYtODJhOC04NGZiODBjYWU3NTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMjRUMTcxNjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OThkMWQyYjg0OWEyYWQyMzQ4YWNhNmU4MGJhMWI1MjlhZjViYjYxYzlhNGY2NDA5NGJiZjUxY2JhOGU0ZDM1NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.1qx7KY7NNUOpciiAjo6E4tnDQYB0JRhm76SnlY8S9CE)

hmm, could be an artifact when I converted that SVG to PNG, not sure. Does the SVG version has it?

Pentarctagon commented:
I'm using a 1440p monitor, which is roughly double the screenshot. Where is the SVG version?

babaissarkar commented:
> SVG as well:
> ![button](https://private-user-images.githubusercontent.com/8469888/398445581-d14f374b-4ec5-4727-8ab0-f5cea9f610f6.svg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUxMDAwNDEsIm5iZiI6MTczNTA5OTc0MSwicGF0aCI6Ii84NDY5ODg4LzM5ODQ0NTU4MS1kMTRmMzc0Yi00ZWM1LTQ3MjctOGFiMC1mNWNlYTlmNjEwZjYuc3ZnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMjVUMDQwOTAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjZkOTk0OGM5NGEyYTRmNGZmMTYyZmI0MDAyMWMzOTcxYjIzMzQxY2JhN2UwZmFhNzdjMjNlYjkwMmNkYTFmMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.tjetiApUUBvprifZD8wUwyJ-F68TmoLHse6JyTZiEG8)

@Pentarctagon ^

Another zipped copy here: [button.svg.zip](https://github.com/user-attachments/files/18243215/button.svg.zip)

Pentarctagon commented:
Well, I don't see it in the SVG no matter how much I zoom in, at least.

knyghtmare reviewed (approved):
I like the new UI appearance, it's sort of on par with the similar genre games of the current decade now.

Vultraz reviewed (approved):
Looks much improved. LGTM
